lib/tictactoe_ui.ex

```diff
@@ -1,88 +1,175 @@
 defmodule TicTacToe.UI do
 
-  def start_game() do
-    TicTacToe.make_move()
-    |> decide_turn_type()
-  end
-  
-  def get_move(game_status) do
-    print_turn(:get_player_move, game_status) 
-  end
+  @clear_screen "\e[H\e[2J"
+
+  def start_game_loop, do: start_game |> game_loop
+
+  def start_game, do: TicTacToe.first_game_status |> do_next_turn
 
-  def get_tile_selection() do
-    IO.gets("Please enter a tile selection:")
-    |> String.trim()
-    |> String.to_integer()
+  def game_loop(:game_over), do: nil
+  def game_loop(game_status) do
+    game_status  
+    |> get_move
+    |> game_loop
   end
 
-  def make_move(selected_tile, game_status) do 
-    TicTacToe.make_move(selected_tile, game_status)
+  def do_turn_type(turn_type, game_status) do
+    IO.puts @clear_screen
+    #print_board(game_status)
+
+    case turn_type do
+      :first_turn -> first_turn(game_status)
+      :bad_input -> bad_input(game_status)
+      :get_player_move -> get_player_move(game_status)
+      :continue -> print_board_and_return_status(game_status)
+      :tile_already_selected -> tile_already_selected(game_status)
+      :win -> win(game_status)
+      :tie -> tie(game_status)
+      _ -> raise "turn type not found"
+    end
   end
 
-   def decide_turn_type(game_status) do
-     elem(game_status, 2)
-     |> print_turn(game_status)
-   end
+  def get_move(game_status), do: do_turn_type(:get_player_move, game_status)
+
+  def make_move(selected_tile, game_status), do: TicTacToe.make_move(selected_tile, game_status)
+
+  def do_next_turn({_,_,turn_type} = game_status) do
+    do_turn_type(turn_type, game_status)
+  end
 
-  def print_turn(:continue, game_status) do
+  def print_board_and_return_status(game_status) do
     print_board(game_status)
     game_status
   end
 
-  def print_turn(:get_player_move, game_status) do
-    get_tile_selection()
-    |> make_move(game_status)
-    |> decide_turn_type()
+  def get_player_move({_, board,_} = game_status) do
+    board_size = Board.size(board)
+    selection = get_tile_selection()
+    
+    if selection == :bad_input or selection > board_size do
+      do_turn_type(:bad_input, game_status)
+    else
+      make_move(selection, game_status)|> do_next_turn()
+    end
   end
 
-  def print_turn(:first_turn, game_status) do
+  def bad_input({player_symbol, _, _}= game_status) do
+    player = Markers.get_player_marker(player_symbol)
+    IO.puts @clear_screen
+    IO.puts "Bad input. Player #{player}, please re-enter selection."
+    print_board(game_status)
+    get_player_move(game_status)
+  end
+
+  def first_turn(game_status) do
     print_board(game_status)
     game_status
   end
 
-  def print_turn(:tile_already_selected, game_status) do
+  def tile_already_selected(game_status) do
     IO.puts "Tile already selected. Please select a different tile."
```

Comment:
We talked about alternatives to the word `tile` in the IPM and at last Friday's demo so please can you update it, Don't forget to update any other references to it in the code (including those mentions of it in the Board module) so that the domain terminology is consistently used amongst the code base.